test/v1/models/NameModelSpec.scala

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models
+
+import play.api.libs.json.{JsObject, JsResultException, Json}
+import support.UnitSpec
+
+class NameModelSpec extends UnitSpec {
+
+  val validJsonWrite: JsObject = Json.obj(
+    "title" -> "MR",
+    "forename" -> "Dovah",
+    "secondForename" -> "Dragon",
+    "surname" -> "Kin",
+    "startDate" -> "2020-10-10",
+    "endDate" -> "2020-10-10"
+  )
+
+  val validJsonRead: JsObject = Json.obj(
+    "title" -> "MR",
+    "forename" -> "Dovah",
+    "secondForename" -> "Dragon",
+    "surname" -> "Kin",
+    "startDate" -> "10-10-2020",
+    "endDate" -> "10-10-2020"
+  )
+
+  val validModel = NameModel(
+    Some("MR"),
+    Some("Dovah"),
+    Some("Dragon"),
+    "Kin",
+    DateModel("10-10-2020"),
+    Some(DateModel("10-10-2020"))
+  )
+
+  "NameModelSpec" should {
+    "correctly parse to Json" in {
+      Json.toJson(validModel) shouldBe validJsonWrite
+    }
+    "correctly parse from Json" in {
+      validJsonRead.as[NameModel] shouldBe validModel
+    }
+
+    "throw an error when parsing from json" when {
+      "the Title field fails validation" in {
```

Comment:
Can you test the validateTitle method for instead of reading it as a model please?